akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/FilteredPayload.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.grpc.internal
+
+import akka.NotUsed
+import akka.annotation.InternalApi
+import akka.dispatch.ExecutionContexts
+import akka.persistence.query.typed.EventEnvelope
+import akka.projection.BySlicesSourceProvider
+import akka.projection.scaladsl.SourceProvider
+import akka.stream.scaladsl.Source
+
+import scala.concurrent.Future
+
+/**
+ * INTERNAL API
+ *
+ * Placeholder object for filtered events when producer event push is used
+ */
+@InternalApi
+private[akka] case object FilteredPayload
+
+/**
+ * INTERNAL API
+ *
+ * Turns envelopes with placeholder events into filtered envelopes on the consuming side of the journal
+ */
+private[akka] final class FilteredPayloadMapper[OffsetType, Event](
```

Comment:
Not sure this idea is good enough, composing these seems to come with surprises. I can't come up with a better way to filter out the `FilteredPayload` entries though. (We need to write them to the journal on the consuming side or else we get sequence numer gaps)

Reply:
Decorating SourceProvider is difficult, I just learned that from https://github.com/akka/akka-projection/pull/934

I don't understand why we need this. Can't we just pass on the EventEnvelope to the EventWriter and let it deal with the filtered event case?

Reply:
May have the same problem as Patrik mentioned wrt Scala/Java DSL as well

Reply:
This is for filtering events out on the consuming side, where the projection reads that placeholder event, that we need to write because non-temporary gaps in sequence numbers breaks the by slice projection.